dartagnan/src/test/java/com/dat3m/dartagnan/c/LLVMTest.java

```diff
@@ -76,11 +82,7 @@ protected long getTimeout() {
     }
 
     protected Provider<Integer> getBoundProvider() {
-        int bound = name == null ? 1 : switch (name) {
-            case LFDS_CHASELEV, LFDS_DGLM, LFDS_HASHTABLE, LFDS_MICHAELSCOTT, LFDS_SAFESTACK, LFDS_TREIBER, LFDS_WSQ -> 2;
-            default -> 1;
-        };
-        return Provider.fromSupplier(() -> bound);
+        return Provider.fromSupplier(() -> bound[0]);
```

Comment:
The issue with your previous code was just that you put the computation of the bound outside of the provider, which caused it to get evaluated at the wrong time. The following would have worked:
```
return () -> {
     return name == null ? 1 : switch (name) {
            case LFDS_CHASELEV, LFDS_DGLM, LFDS_HASHTABLE, LFDS_MICHAELSCOTT, LFDS_SAFESTACK, LFDS_TREIBER, LFDS_WSQ -> 2;
            default -> 1;
        };
}
```

Also, in your newer version I don't think the array is needed. A simple `int` is enough.